zio-http/src/main/scala/zio/http/Body.scala

```diff
@@ -41,6 +43,23 @@ trait Body { self =>
    */
   def ++(that: Body): Body = if (that.isEmpty) self else that
 
+  /**
+   * Decodes the content of the body as a value based on a zio-schema
+   * [[zio.schema.codec.BinaryCodec]].<br>
+   *
+   * Example for json:
+   * {{{
+   * import zio.schema.json.codec._
+   * case class Person(name: String, age: Int)
+   * implicit val schema: Schema[Person] = DeriveSchema.gen[Person]
+   * val person = Person("John", 42)
+   * val body = Body.from(person)
+   * val decodedPerson = body.as[Person]
+   * }}}
+   */
+  def as[A](implicit codec: BinaryCodec[A], trace: Trace): Task[A] =
```

Comment:
Should this be `to` or `into`? `as` has a meaning in some ZIO contexts as `map(_ => a)`.

Reply:
I chose the name because it is called `queryAs` if you want to get a typed query param

Reply:
Actually I am fine with both. I can change it next time a PC is an reach

Reply:
Maybe that should be `queryTo` or `queryInto` for consistency?

Reply:
Should it then also be `body.toString` and not `body.asString`? But this has an obvious conflict.

Reply:
Agreed, please update any other cases like this that you find.